### PR TITLE
fix: set AuthURLParam to set client id and secret

### PR DESCRIPTION
**Pull request by shipperizer**
workaround to deal with #golang/oauth2/320

### CODE REVIEW

File: internal/broker/broker.go [hunk 1]
```diff
@@ -320,7 +320,15 @@ func (b *Broker) generateUILayout(session *sessionInfo, authModeID string) (map[
...
-		response, err := session.authCfg.oauth.DeviceAuth(ctx)
+
+		var opt oauth2.AuthCodeOption
+
+		if secret := session.authCfg.oauth.ClientSecret; secret != "" {
+			opt = oauth2.SetAuthURLParam("client_secret", secret)
+		}
+
+		response, err := session.authCfg.oauth.DeviceAuth(ctx, opt)
+
```
Comment (3v1n0) [inline internal/broker/broker.go:331]: Just a nit:

```suggestion

		var authOpt oauth2.AuthCodeOption
		if secret := session.authCfg.oauth.ClientSecret; secret != "" {
			authOpt = oauth2.SetAuthURLParam("client_secret", secret)
		}

		response, err := session.authCfg.oauth.DeviceAuth(ctx, authOpt)
```

**Review by 3v1n0 — APPROVED** [review]

**Comment by adombeck** [comment]
I would also appreciate an actual link to the issue instead of #golang/oauth2/320. Thanks!

**Comment by 3v1n0** [comment]
> I would also appreciate an actual link to the issue instead of #golang/oauth2/320. Thanks!

Yeah, I wanted to mention that too for us old school `git log` users :joy:

**Comment by didrocks** [comment]
Now that https://github.com/ubuntu/authd-oidc-brokers/pull/178 is merged, do you mind rebasing on it and target the main branch directly?

Also, a small comment explaining which providers needs this and the details would be great too! Thanks :)

**Pull request closed without merging**